server/resp/src/main/java/org/infinispan/server/resp/commands/connection/HELLO.java

```diff
@@ -35,25 +49,33 @@ public CompletionStage<RespRequestHandler> perform(Resp3AuthHandler handler,
       String version = new String(respProtocolBytes, CharsetUtil.UTF_8);
       if (!version.equals("3")) {
          ByteBufferUtils.stringToByteBufAscii("-NOPROTO sorry this protocol version is not supported\r\n", handler.allocator());
+         return handler.myStage();
+      }
+
+      if (arguments.size() == 4) {
+         successStage = handler.performAuth(ctx, arguments.get(2), arguments.get(3));
+      } else if (!handler.isAuthorized() && handler.canUseCertAuth()) {
+         successStage = handler.performAuth(ctx);
       } else {
-         if (arguments.size() == 4) {
-            successStage = handler.performAuth(ctx, arguments.get(2), arguments.get(3));
-         } else if (!handler.isAuthorized() && handler.canUseCertAuth()) {
-            successStage = handler.performAuth(ctx);
+         if (handler.isAuthenticationRequired()) {
+            RespErrorUtil.customRawError("NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO <proto> AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time", handler.allocator());
```

Comment:
Is this the error that REDIS returns? I thought they normally have a dash as the first character?